multi_partner_learning.py

```diff
@@ -124,28 +193,39 @@ def compute_test_score(self):
 
         x_val, y_val = self.val_data
         x_test, y_test = self.test_data
-
+        # Prepare the input data if the learning approach is stacking
+        if self.learning_approach == "stacking":
+            x_val_extended = [x_val for _ in range(partners_count)]
+            x_val=x_val_extended
+            x_test_extended = [x_test for _ in range(partners_count)] 
+            x_test=x_test_extended
+            
         # First, if only one partner, fall back to dedicated single partner function
         if partners_count == 1:
             return self.compute_test_score_for_single_partner(partners_list[0])
 
         # Else, continue onto a federated learning procedure
         partners_list = sorted(partners_list, key=operator.attrgetter("id"))
         logger.info(
-            f"## Training and evaluating model on partners with ids: {['#' + str(p.id) for p in partners_list]}")
+            f"## Training and evaluating model on partners with ids: {['#' + str(p.id) for p in partners_list]}"
+        )
 
         # Initialize variables
-        model_to_evaluate, sequentially_trained_model = None, None
-        if self.learning_approach in ['seq-pure', 'seq-with-final-agg']:
+        model_to_evaluate, sequentially_trained_model, stacking_model = None, None, None
```

Comment:
Here we could keep only one of the two (`sequentially_trained_mode` and `stacking_model`), as only one of them is used downstream; it would require giving it a more generic name, in substance "the model that we use across multiple epochs"). But would that really be better, I am not too sure for readibility. What do you think @Thomas-Galtier @RomainGoussault @natct10 ?